test/common/upstream/load_balancer_simulation_test.cc

```diff
@@ -32,6 +32,54 @@ static HostSharedPtr newTestHost(Upstream::ClusterInfoConstSharedPtr cluster,
                    envoy::api::v2::endpoint::Endpoint::HealthCheckConfig::default_instance())};
 }
 
+// Simulate weighted LR load balancer.
+TEST(DISABLED_LeastRequestLoadBalancerWeightTest, Weight) {
+  const uint64_t num_hosts = 4;
+  const uint64_t weighted_subset_percent = 50;
+  const uint64_t weight = 2;
+
+  PrioritySetImpl priority_set_;
+  std::shared_ptr<MockClusterInfo> info_{new NiceMock<MockClusterInfo>()};
+  HostSet& host_set = priority_set_.getOrCreateHostSet(0);
+  HostVector hosts;
+  for (uint64_t i = 0; i < num_hosts; i++) {
+    const bool should_weight = i < num_hosts * (weighted_subset_percent / 100.0);
+    hosts.push_back(makeTestHost(info_, fmt::format("tcp://10.0.{}.{}:6379", i / 256, i % 256),
+                                 should_weight ? weight : 1));
+  }
+  HostVectorConstSharedPtr updated_hosts{new HostVector(hosts)};
+  HostsPerLocalitySharedPtr updated_locality_hosts{new HostsPerLocalityImpl(hosts)};
+  host_set.updateHosts(updated_hosts, updated_hosts, updated_locality_hosts, updated_locality_hosts,
+                       {}, hosts, {});
+
+  Stats::IsolatedStoreImpl stats_store_;
+  ClusterStats stats_{ClusterInfoImpl::generateStats(stats_store_)};
+  stats_.max_host_weight_.set(weight);
+  NiceMock<Runtime::MockLoader> runtime_;
+  Runtime::RandomGeneratorImpl random_;
+  envoy::api::v2::Cluster::CommonLbConfig common_config_;
+  LeastRequestLoadBalancer lb_{priority_set_, nullptr, stats_, runtime_, random_, common_config_};
+
+  std::unordered_map<HostConstSharedPtr, uint64_t> host_hits;
+  const uint64_t total_requests = 100;
+  for (uint64_t i = 0; i < total_requests; i++) {
+    host_hits[lb_.chooseHost(nullptr)]++;
+  }
+
+  std::unordered_map<uint64_t, double> weight_to_percent;
+  for (const auto& host : host_hits) {
+    std::cout << fmt::format("url:{}, weight:{}, hits:{}, percent_of_total:{}\n",
+                             host.first->address()->asString(), host.first->weight(), host.second,
+                             (static_cast<double>(host.second) / total_requests) * 100);
+    weight_to_percent[host.first->weight()] +=
+        (static_cast<double>(host.second) / total_requests) * 100;
+  }
+
+  for (const auto& weight : weight_to_percent) {
+    std::cout << fmt::format("weight:{}, percent:{}\n", weight.first, weight.second);
+  }
+}
+
```

Comment:
Does this test capture the active request aspect of the LB?

Reply:
It doesn't currently, I did do it locally when I was playing around. I can add something back with comments.